contracts/eden/src/inductions.cpp

```diff
@@ -105,13 +110,112 @@ namespace eden
       return induction_tb.get(id, "unable to find induction");
    }
 
+   const induction& inductions::get_endorsed_induction(eosio::name invitee) const
+   {
+      endorsed_induction_table_type endorsed_induction_tb(contract, default_scope);
+      return get_induction(endorsed_induction_tb.get(invitee.value).induction_id);
+   }
+
    void inductions::check_valid_induction(const induction& induction) const
    {
       auto induction_lifetime = eosio::current_time_point() - induction.created_at.to_time_point();
       eosio::check(induction_lifetime.to_seconds() <= induction_expiration_secs,
                    "induction has expired");
    }
 
+   void inductions::update_video(const induction& induction,
+                                 const std::string& video)
+   {
+      check_valid_induction(induction);
+      validate_video(video);
+
+      induction_tb.modify(induction_tb.iterator_to(induction), eosio::same_payer,
+                          [&](auto& row) { row.video = video; });
+
+      reset_endorsements(induction.id);
+   }
+
+   void inductions::endorse(const induction& induction, eosio::name account, eosio::checksum256 induction_data_hash)
+   {
+      check_valid_induction(induction);
+      eosio::check(!induction.video.empty(), "Video not set");
+      eosio::check(!induction.new_member_profile.name.empty(), "Profile not set");
```

Comment:
I think that extracting this to `validate_completed_induction()` could be better -- this could call the `check_valid_induction` inside plus all the checks related to endorsement. thoughts?

Reply:
It would make sense if we needed the same checks elsewhere.  As it is, I don't see the point.